test/forge/BlueTest.sol

```diff
@@ -0,0 +1,129 @@
+        morpho.enableLltv(DEFAULT_TEST_LLTV);
+        vm.stopPrank();
+
+        vm.prank(SUPPLIER);
+        loanToken.approve(address(morpho), type(uint256).max);
+
+        vm.startPrank(BORROWER);
+        loanToken.approve(address(morpho), type(uint256).max);
+        collateralToken.approve(address(morpho), type(uint256).max);
+        vm.stopPrank();
+    }
+
+    /* TESTS */
+
+    function testAdaptiveCurveIrm(uint256 amountSupplied, uint256 amountBorrowed, uint256 timeElapsed) public {
+        amountSupplied = bound(amountSupplied, MIN_TEST_AMOUNT, MAX_TEST_AMOUNT);
+        amountBorrowed = bound(amountBorrowed, MIN_TEST_AMOUNT, amountSupplied);
+        timeElapsed = bound(timeElapsed, MIN_TIME_ELAPSED, MAX_TIME_ELAPSED);
+
+        MarketParams memory marketParams = MarketParams(
+            address(loanToken), address(collateralToken), address(oracle), address(adaptiveCurveIrm), DEFAULT_TEST_LLTV
+        );
+        morpho.createMarket(marketParams);
+
+        loanToken.setBalance(SUPPLIER, amountSupplied);
+        vm.prank(SUPPLIER);
+        morpho.supply(marketParams, amountSupplied, 0, SUPPLIER, hex"");
+
+        uint256 collateralAmount = amountBorrowed.wDivUp(DEFAULT_TEST_LLTV);
+        collateralToken.setBalance(BORROWER, collateralAmount);
+
+        vm.startPrank(BORROWER);
+        morpho.supplyCollateral(marketParams, collateralAmount, BORROWER, hex"");
+        morpho.borrow(marketParams, amountBorrowed, 0, BORROWER, BORROWER);
+        vm.stopPrank();
+
+        vm.warp(timeElapsed);
+
+        morpho.accrueInterest(marketParams);
+    }
+
+    function testFixedRateIrm(uint256 amountSupplied, uint256 amountBorrowed, uint256 fixedRate, uint256 timeElapsed)
+        public
+    {
+        amountSupplied = bound(amountSupplied, MIN_TEST_AMOUNT, MAX_TEST_AMOUNT);
+        amountBorrowed = bound(amountBorrowed, MIN_TEST_AMOUNT, amountSupplied);
+        fixedRate = bound(timeElapsed, 1, fixedRateIrm.MAX_BORROW_RATE());
+        timeElapsed = bound(timeElapsed, MIN_TIME_ELAPSED, MAX_TIME_ELAPSED);
+
+        MarketParams memory marketParams = MarketParams(
+            address(loanToken), address(collateralToken), address(oracle), address(fixedRateIrm), DEFAULT_TEST_LLTV
+        );
+        fixedRateIrm.setBorrowRate(marketParams.id(), fixedRate);
+        morpho.createMarket(marketParams);
+
+        loanToken.setBalance(SUPPLIER, amountSupplied);
+        vm.prank(SUPPLIER);
+        morpho.supply(marketParams, amountSupplied, 0, SUPPLIER, hex"");
+
+        uint256 collateralAmount = amountBorrowed.wDivUp(DEFAULT_TEST_LLTV);
+        collateralToken.setBalance(BORROWER, collateralAmount);
+
+        vm.startPrank(BORROWER);
+        morpho.supplyCollateral(marketParams, collateralAmount, BORROWER, hex"");
+        morpho.borrow(marketParams, amountBorrowed, 0, BORROWER, BORROWER);
+        vm.stopPrank();
+
+        vm.warp(timeElapsed);
+
+        morpho.accrueInterest(marketParams);
```

Comment:
It would make sense to test that the accrued intereste are in a certain range (approximate it very roughly). This is because forgetting the line `fixedRateIrm.setBorrowRate` would still make the test pass, which is a bit loose IMO.
Could do the same thing for the adaptive curve IRM, although finding a rough estimate is a bit harder